connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2585,10 +2586,10 @@ class SparkConnectPlanner(
             proto.LocalRelation
               .newBuilder()
               .setData(ByteString.copyFrom(bytes))))
+      metrics.addAllMetrics(MetricGenerator.transformPlan(df).asJava)
```

Comment:
Could you add comment why this is done. Some thing like "This is a command like `show tables` and we need to eagerly execute and return the results". 
Most readers (like me) wouldn't know.
Also why are results called metrics?

Reply:
I added the comment on the conditional below to explain why we're skipping metrics.